crates/sui/src/sui_commands.rs

```diff
@@ -112,6 +113,7 @@ pub enum SuiCommand {
 
 impl SuiCommand {
     pub async fn execute(self) -> Result<(), anyhow::Error> {
+        move_package::package_hooks::register_package_hooks(Box::new(SuiPackageHooks {}));
```

Comment:
This is the line needed to register our field. `SuiCommand::execute` is the common caller of `move build` and `sui client publish`, so I put it here (it's also possible to put it in main ([example](https://github.com/aptos-labs/aptos-core/blob/ee8e02bec85a8e2ea7f1a24e8c53acdc492ad6cb/crates/aptos/src/main.rs#L19)), or duplicate in each branch if we don't want to run this unconditionally)

Reply:
I think this is a fine place to put it when it comes to our CLI commands, but we will also have to think about for tests (we have tests that need to publish packages).  Is there a way to cater to those as well?  There may not be, in which case we'll just need to be careful about adding the hooks in all the right places.